FlowCryptCommon/Extensions/OptionalExtensions.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  OptionalExtensions.swift
+//  FlowCryptCommon
+//
+//  Created by Ioan Moldovan on 5/5/22
+//  Copyright © 2017-present FlowCrypt a. s. All rights reserved.
+//
+
+import Foundation
+
+extension Optional {
+    public func ifNotNil<U>(_ transform: (Wrapped) throws -> U) rethrows -> U? {
+        switch self {
+        case .some(let value):
+          return .some(try transform(value))
```

Comment:
@ioanmo226 is `try` needed here?
It looks like `return .some(transform(value))` should work well here

Reply:
![image](https://user-images.githubusercontent.com/77344191/167011282-ed989971-c7c8-45f0-8874-3dddc8c1758b.png)

transform might throw error

Reply:
It's because you marked it with `throws` in method description.
Xcode won't give any errors after removing `throws` there:
```swift
public func ifNotNil<U>(_ transform: (Wrapped) -> U) -> U? {
        switch self {
        case .some(let value):
          return .some(transform(value))
        case .none:
          return .none
        }
    }
```

Reply:
yeah, correct. because closure inside ifNotNil might throw error

Reply:
we may want to do something like `expiration.ifNotNil { try getSomething($0) }`

Reply:
Ah, got it, then all good 👍